stories/upload-csv-popup.stories.js

```diff
@@ -0,0 +1,219 @@
+Leja Arroyo,leja.arroyo@random.com,Random Office`,
+    ],
+    'all new data file.csv'
+)
+
+const existingTableOldEmailArgs = {
+    defaultValue: {
+        '61e34ba4dd28d45f2c6c66be': {
+            uniqueId: '61e34ba4dd28d45f2c6c66be',
+            name: 'Diana Russel',
+            email: 'my.old.email@example.com',
+            office: 'old office',
+        },
+    },
+}
+
+const existingTableNewEmailArgs = {
+    defaultValue: {
+        '61e34ba4dd28d45f2c6c66be': {
+            uniqueId: '61e34ba4dd28d45f2c6c66be',
+            name: 'Diana Russel',
+            email: 'my.new.email@example.com',
+            office: 'old office',
+        },
+    },
+}
+
+const Template = (args, context) => {
+    const { electionOM } = context
+    const { defaultValue, ...otherArgs } = args
+    const [electionObj, electionMethods] = electionOM
+    useEffect(() => defaultValue && electionMethods.upsert({ candidates: defaultValue }), [defaultValue])
+
+    return (
+        <UploadCSVPopup
+            visible='true'
+            electionObj={electionObj}
+            electionMethods={electionMethods}
+            closePopup={() => console.log('close popup called')}
+            {...otherArgs}
+        />
+    )
+}
+
+export const Default = Template.bind({})
+
+export const EmptyTableNoUniqueIds = Template.bind({})
+EmptyTableNoUniqueIds.args = { defaultValue: {} }
+EmptyTableNoUniqueIds.play = async ({ canvasElement }) => {
+    const canvas = within(canvasElement)
+    await new Promise(r => setTimeout(r, 1000))
+
+    await userEvent.upload(canvas.getByTestId('file-select-input'), noUniqueIdsFile)
+    await new Promise(r => setTimeout(r, 1000))
+
+    await userEvent.click(canvas.getByText('Extract Data'))
+}
+
+export const EmptyTableWithUniqueIds = Template.bind({})
+EmptyTableWithUniqueIds.args = { defaultValue: {} }
+EmptyTableWithUniqueIds.play = async ({ canvasElement }) => {
+    const canvas = within(canvasElement)
+    await new Promise(r => setTimeout(r, 1000))
+
+    await userEvent.upload(canvas.getByTestId('file-select-input'), withUniqueIdsFile)
+    await new Promise(r => setTimeout(r, 1000))
+
+    await userEvent.click(canvas.getByText('Extract Data'))
+}
+
+export const ExistingTableWithUniqueIds = Template.bind({})
+ExistingTableWithUniqueIds.args = { ...existingTableOldEmailArgs }
+ExistingTableWithUniqueIds.play = async ({ canvasElement }) => {
+    const canvas = within(canvasElement)
+    await new Promise(r => setTimeout(r, 1000))
+
+    await userEvent.upload(canvas.getByTestId('file-select-input'), withUniqueIdsFile)
+    await new Promise(r => setTimeout(r, 1000))
+
+    await userEvent.click(canvas.getByText('Extract Data'))
+}
+
+export const ExistingTableMatchEmail = Template.bind({})
+ExistingTableMatchEmail.args = { ...existingTableNewEmailArgs }
+ExistingTableMatchEmail.play = async ({ canvasElement }) => {
+    const canvas = within(canvasElement)
+    await new Promise(r => setTimeout(r, 1000))
```

Comment:
In these tests, am I suppose to have a csv file or something to upload.  A demo of this might help, or we may have to find a way to add csv files for testing somewhere in the repo.  Not sure if there are differences between your environment and mine that make a difference.

Reply:
There's an issue with these where the test uploads a file and the story continues in the background, but the browser opens up the "browse for file" dialogue. The story itself still works from what I saw on my end, and I spent some time trying to get the dialogue to go away but didn't want to spend a ton of time on it since the functionality was there and it was an issue specific to storybook. I'll do another pass through on it to see if I can find a better solution.

Reply:
Actually, I'm rerunning some of these they are no longer updating the electionObj so something's definitely not working quite right. I swear this was just working... I'll get that fixed up too.

Reply:
I do see these stories working, but the dialog popping up appears to be a defect or issue within storybook from what I can tell. I will create an issue on their github page in the meantime, and add a decorator to the stories to explain this behavior to future people using these stories.